gammapy/scripts/image_pipe.py

```diff
@@ -21,37 +28,152 @@ def image_pipe_main(args=None):
 
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    center : `astropy.coordinates.SkyCoord`
+        Center of the image
+    energy_band : `~gammapy.utils.energy.Energy
+        Energy band for which we want to compute the image
+    offset_band :`astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, center, energy_band, offset_band, data_store, counts_image=None, bkg_image=None):
+        self.data_store = data_store
+        self.center = center
+        self.energy_band = energy_band
+        self.offset_band = offset_band
+        if not counts_image:
+            self.counts_image = SkyMap.empty(nxpix=1000, nypix=1000, binsz=0.01, xref=self.center.l.deg,
```

Comment:
I would suggest to only to initialize an empty `self.maps = SkyMapCollection()` and add the (i.e. counts, background, exposure etc. step by step to this container in the corresponding methods). The binning information (nxpix, nypix, binsz, etc.) shouldn't be hard-coded but rather passed to `self.make_counts` or be stored as a dictionary. To avoid repeating the binning information all the time you can use `SkyMap.empty_like()`.